qiskit/visualization/dag_visualization.py

```diff
@@ -166,7 +224,14 @@ def edge_attr_func(edge):
     dot = pydot.graph_from_dot_data(dot_str)[0]
 
     if filename:
+        if "." not in filename:
+            raise InvalidFileError("Parameter 'filename' must be in format 'name.extension'")
         extension = filename.split(".")[-1]
+        if extension not in FILENAME_EXTENSIONS:
+            raise InvalidFileError(
+                "Filename extension must be one of: "
+                + " ".join([str(elem) for elem in FILENAME_EXTENSIONS])
```

Comment:
All the elements in `FILENAME_EXTENSIONS` are already strings, so this is just the same as `" ".join(FILENAME_EXTENSIONS)`.